### ISSUE TITLE
Update to support configuring the "run" option for the language server

### ISSUE THREAD

**Issue by nrayburn-tech** [oxc-project/oxc-intellij-plugin#62]
1. Add a new property to OxcSettingsComponent and OxcSettingsState.
2. Add the new input to OxcSettingsConfigurable (probably a Combo Box, but radio buttons may be fine too).
3. Update OxcLspServerDescriptor#createInitializationOptions to send the configured "run" option to the language server.

**Comment by nrayburn-tech** [comment]
IntelliJ LSP doesn't currently support the `textDocument/didSave` event.
- Option 1, implement it within the plugin by listening for document saves and sending the event ourselves.  Not sure how difficult this would be.
- Option 2, wait for IntelliJ support https://youtrack.jetbrains.com/issue/IJPL-156726/Support-for-textDocument-didSave.

**Comment by nrayburn-tech** [comment]
JetBrains is adding support in 2024.3.1. This brings up the question of what JetBrains IDE versions this plugin should target for support. Maybe just the latest stable?
The LSP APIs being used are experimental and still under active development, so supporting additional versions isn’t going to be easy.

**Comment by nrayburn-tech** [comment]
@Boshen @IWANABETHATGUY  (or anybody else) have any thoughts on what versions of JetBrains IDEs to try and support?  Given my previous comment about the LSP APIs being experimental, I think it only makes sense to support the latest version for now.

**Comment by IWANABETHATGUY** [comment]
I am not familiar with JetBrains actually, can you have a look at the biome? Maybe same as biome should be enough.

**Comment by nrayburn-tech** [comment]
A lot of information below, tried to separate it into reasonable chunks.

### Biome and JetBrains Statistics

Biome is currently targeting the [latest release](https://github.com/biomejs/biome-intellij/blob/main/gradle.properties#L8-L14). 2024.3.* (or 243.* in the other version format in use some places).

https://plugins.jetbrains.com/docs/marketplace/product-versions-in-use-statistics.html contains some 2023 statistics from JetBrains on which versions were in use at the time.  It indicates targeting the current and prior 2 major releases should cover 80% of the JetBrains users.

### Difficulty with Multiple Versions

The difficulty with targeting multiple versions is the major versions may have API changes, especially with the LSP API being experimental.  It is possible to release a different plugin version targeting different JetBrains versions, but it is not trivial.  For example, we might have to maintain multiple branches corresponding to the different releases that compatibility is desired for.

An explanation of an example branching strategy that could be used to support multiple IDE versions.
1. Branch `2024.1` would target IDE versions `2024.1` or newer.
1. IDE version `2024.3` is released and contains API changes that we need to adopt for whatever reason (incompatible with our existing code or contains new APIs we want to use)
1. We create a new `2024.3` branch for `2024.3` or newer.
1. The 2024.1 branch now supports 2024.1 and 2024.2 due to the `2024.3` branch being created as the latest.

This would mean maintaining changes in each branch.  Depending on the significance of the API changes, the effort here could vary greatly.

### Suggestion

Target latest for now as we have a need to change.  As the LSP API stabilizes and the plugin stops adding new features, expand the supported versions to cover at least the last 3 major releases.

**Comment by IWANABETHATGUY** [comment]
Yeah, that should make sense, let's apply your suggestion, and wait user's feedback
